src/unix/darwin.c

```diff
@@ -65,28 +65,33 @@ uint64_t uv__hrtime(uv_clocktype_t type) {
 
 
 int uv_exepath(char* buffer, size_t* size) {
-  uint32_t usize;
-  int result;
-  char* path;
-  char* fullpath;
+  /* realpath(exepath) may be > PATH_MAX so double it to be on the safe side. */
+  char abspath[PATH_MAX * 2 + 1];
```

Comment:
Probably, but I'm more comfortable over-allocating a little.  I checked the realpath() implementation in libc and it's a mess; realpath() shouldn't go beyond PATH_MAX but the OS X implementation does.  If you pass NULL as the second argument, it happily scribbles beyond the heap memory it allocates...  Ergo, I figured I'd err on the side of caution and add a good amount of slop.